src/library.js

```diff
@@ -4244,7 +4254,11 @@ LibraryManager.library = {
     return [args, funcname, str];
   },
 
-  emscripten_get_callstack_js__deps: ['_emscripten_traverse_stack'],
+  emscripten_get_callstack_js__deps: ['_emscripten_traverse_stack'
+#if MINIMAL_RUNTIME
+  , '$jsStackTrace', '$warnOnce', '$demangle'
+#endif
```

Comment:
I'm unhappy about changes like this - do we need more code like this that is specific to the minimal runtime? Can we not use the same new deps in the other path as well, keeping things simple?

Reply:
In scenarios where I could use the deps for both runtime modes, I would - that would definitely be the simples.

However it was too difficult to move those functions to JS library. The difference is that in MINIMAL_RUNTIME, there is no "untracked" JS runtime code depends on these, but in traditional runtime, there is JS code in the runtime files that call into these functions, making it infeasible to also directly move these to become JS library functions in traditional runtime, without more detailed refactor.

Another aspect is that I want to avoid being in a position where I introduce breaking changes to other Emscripten users, since I am not in as comfortable position any more to provide support to breakages. That is why I try to pay attention to keep things working for users as-is where possible. Taking these functions away from the core runtime will break users that are calling them in their code (although adapting will be moderately easy)

Ultimately I still hope that the traditional runtime would wither away, as we learn how to re-model all code using better linking aware JS lib approaches, and at that point these kinds of differences could become obsolete?